.github/workflows/build_ut.yml

```diff
@@ -118,26 +141,27 @@ jobs:
           branch: master
           name: coverage_report_stats
           path: master_artifacts
-
+      
       # Run the process_coverage_stats script
       - name: Process Coverage Statistics
         if: ${{ success() && github.ref != 'refs/heads/master' }}
         run: python scripts/coverage/process_coverage_stats.py ./master_artifacts/coverage_statistics.txt build/coverage_statistics.txt
-
+      
       # Get process_coverage_stats script output
       - id: get-comment-body
-        if: ${{ success() && github.ref != 'refs/heads/master' }}
+        if: ${{ (success() || failure()) && github.ref != 'refs/heads/master' }}
```

Comment:
Is that comparison needed?

Reply:
FYI, I think by default, test steps are skipped if status is failure or cancelled, but maybe make sure that's the case.

Reply:
Yup, just tested it out on a different branch and Luke is right, test steps are skipped if the status is failure or cancelled, so both are comparisons are required